source/common/ssl/context_impl.cc

```diff
@@ -137,35 +147,53 @@ bssl::UniquePtr<SSL> ContextImpl::newSsl() const {
   return bssl::UniquePtr<SSL>(SSL_new(ctx_.get()));
 }
 
-bool ContextImpl::verifyPeer(SSL* ssl) const {
-  bool verified = true;
+int ContextImpl::verifyCallback(int ok, X509_STORE_CTX* store) {
```

Comment:
Yeah, I don't have strong opinions either way. The higher-level callback "makes more sense" to me, but like I said, I don't understand this X.509 stack anyway. :-) You're right that the eventual transition will want a tweak here anyway, but that's sufficiently far into the future that it's unclear how to coherently prepare for it.

(Though I see you've switched it to the other callback here. *shrug* whatever works.)